chore(docs): add Jina Deployment docs (#6079)
jina-ai/jina: docs/concepts/jcloud/index.md

:width: 90%
```

## Deployments

### Deploy

> A Jina [Deployment](https://docs.jina.ai/concepts/orchestration/Deployment/) represents an end-to-end task such as indexing, searching, or recommending. In this document, we use "project" and "Deployment" interchangeably.

Reviewer: This is not the same description, I do not think we even need to define this

> In Jina's idiom, a project is a [Deployment](https://docs.jina.ai/concepts/orchestration/Deployment/), which represents an end-to-end task such as indexing, searching or recommending. In this document, we use "project" and "Deployment" interchangeably.

Reviewer: Avoid re-defining the term, just act as it is already defined.

Reviewer 2: ```suggestion
A Jina [Deployment](https://docs.jina.ai/concepts/orchestration/Deployment/) represents an end-to-end task such as indexing, searching, or recommending. In this document, we use "project" and "Deployment" interchangeably.
```

```{caution}
When `jcloud` deploys a deployment it automatically appends the following global arguments to the `deployment.yml`, if not present:
```

```yaml
jcloud:
  version: jina-version
  docarray: docarray-version
```

#### Single YAML file

> A self-contained YAML file, consisting of all configuration at the [Deployment](https://docs.jina.ai/concepts/orchestration/deployment/)-level and [Executor](https://docs.jina.ai/concepts/serving/executor/)-level.

Reviewer 2: ```suggestion
A self-contained YAML file, consisting of all configuration information at the [Deployment](https://docs.jina.ai/concepts/orchestration/deployment/)-level and [Executor](https://docs.jina.ai/concepts/serving/executor/)-level.
```

> > A Deployment's `uses` must follow the format `jinaai+docker://<username>/MyExecutor` (from [Executor Hub](https://cloud.jina.ai)) to avoid any local file dependencies:

Reviewer 2: ```suggestion
> A Deployment's `uses` parameter must follow the format `jinaai+docker://<username>/MyExecutor` (from [Executor Hub](https://cloud.jina.ai)) to avoid any local file dependencies:
```

```yaml
# deployment.yml
jtype: Deployment
with:
  protocol: grpc
  uses: jinaai+docker://jina-ai/Sentencizer
```

To deploy:

```bash
jc deployment deploy ./deployment.yaml
```

The Deployment is successfully deployed when you see:

```{figure} img/deployment/deploy.png
:width: 70%
```
---

> You will get a Deployment ID, say `pretty-monster-130a5ac952`. This ID is required to manage, view logs and remove the Deployment.

Reviewer 2: ```suggestion
You will get a Deployment ID, for example, `pretty-monster-130a5ac952`. This ID is required to manage, view logs, and remove the Deployment.
```

> As this Deployment is deployed with the default gRPC protocol (feel free to change it to `http`), you can use `jina.Client` to access it:

Reviewer 2: ```suggestion
Since this Deployment is deployed with the default gRPC protocol (feel free to change it to `http`), you can use `jina.Client` to access it:
```

```python
from jina import Client, Document

print(
    Client(host='grpcs://executor-pretty-monster-130a5ac952.wolf.jina.ai').post(
        on='/', inputs=Document(text='hello')
    )
)
```

(jcloud-deployoment-status)=
### Get status

To get the status of a Deployment:
```bash
jc deployment status pretty-monster-130a5ac952
```

```{figure} img/deployment/status.png
:width: 70%
```

### List Deployments

To list all of your "Starting", "Serving", "Failed", "Updating", and "Paused" Deployments:

```bash
jc deployment list
```

```{figure} img/deployment/list.png
:width: 90%
```

You can also filter your Deployments by passing a phase:

```bash
jc deployment list --phase Deleted
```

```{figure} img/deployment/list_deleted.png
:width: 90%
```

Or see all Deployments:

```bash
jc deployment list --phase all
```

```{figure} img/deployment/list_all.png
:width: 90%
```

### Remove Deployments

> You can remove a single Deployment, multiple Deployments or even all Deployments by passing different identifiers.

Reviewer 2: ```suggestion
You can remove a single Deployment, multiple Deployments, or even all Deployments by passing different commands to the `jc` executable at the command line.
```

To remove a single Deployment:

```bash
jc deployment remove pretty-monster-130a5ac952
```

To remove multiple Deployments:

```bash
jc deployment remove pretty-monster-130a5ac952 artistic-tuna-ab154c4dcc
```

To remove all Deployments:

```bash
jc deployment remove all
```

> By default, removing multiple or all Deployments is an interactive process where you must give confirmation before each Deployment is deleted. To make it non-interactive, set the below environment variable before running the command:

Reviewer 2: ```suggestion
By default, removing all or multiple Deployments is an interactive process where you must give confirmation before each Deployment is deleted. To make it non-interactive, set the below environment variable before running the command:
```
